executors/cmdexec/brick.go

```diff
@@ -142,13 +131,74 @@ func (s *CmdExecutor) BrickDestroy(host string,
 	}
 	tp := output[0]
 
+	return dev, tp, nil
+}
+
+func (s *CmdExecutor) deleteBrickLV(host, lv string) error {
+	// Remove the LV (by device name)
+	commands := []string{
+		fmt.Sprintf("lvremove --autobackup=%v -f %v",
+			utils.BoolToYN(s.BackupLVM), lv),
+	}
+	_, err := s.RemoteExecutor.RemoteCommandExecute(host, commands, 5)
+	return err
+}
+
+func (s *CmdExecutor) countThinLVsInPool(host, tp string) (int, error) {
+	// Detect the number of bricks using the thin-pool
+	commands := []string{
+		fmt.Sprintf("lvs --noheadings --options=thin_count %v", tp),
+	}
+	output, err := s.RemoteExecutor.RemoteCommandExecute(host, commands, 5)
+	if err != nil {
+		logger.Err(err)
+		return 0, fmt.Errorf("Unable to determine number of logical volumes in "+
+			"thin pool %v on host %v", tp, host)
+	}
+	thin_count, err := strconv.Atoi(strings.TrimSpace(output[0]))
+	if err != nil {
+		return 0, fmt.Errorf("Failed to convert number of logical volumes in thin pool %v on host %v: %v", tp, host, err)
+	}
+	return thin_count, nil
+}
+
+func (s *CmdExecutor) BrickDestroy(host string,
+	brick *executors.BrickRequest) (bool, error) {
+
+	godbc.Require(brick != nil)
+	godbc.Require(host != "")
+	godbc.Require(brick.Name != "")
+	godbc.Require(brick.VgId != "")
+	godbc.Require(brick.Path != "")
+	godbc.Require(brick.TpName != "")
+	godbc.Require(brick.LvName != "")
+
+	var (
+		umountErr      error
+		spaceReclaimed bool
+	)
+
+	// TODO: convert to a best effort sanity check
+	// dev, tp, err := s.brickStorage(host, brick)
+	// if err != nil {
+	// 	return spaceReclaimed, err
+	// }
+
 	// Try to unmount first
-	commands = []string{
+	commands := []string{
 		fmt.Sprintf("umount %v", brick.Path),
 	}
 	_, umountErr = s.RemoteExecutor.RemoteCommandExecute(host, commands, 5)
 	if umountErr != nil {
-		logger.Err(err)
+		logger.Err(umountErr)
+		// check if the brick was previously unmounted
+		out, e := s.RemoteExecutor.RemoteCommandExecute(
+			host, []string{"mount"}, 5)
+		if e == nil && len(out) == 1 && !strings.Contains(out[0], brick.Path) {
```

Comment:
Why `len(out) == 1`? there could be more mounts.
Did you intend to use some form of `mount | grep ...` like in other places?

Reply:
The length of out in this case is the length of the slice passed to RemoteCommandExecute not the number of lines in the output of the mount command. The check is somewhat redundant but it is basically asserting that only the mount command was run by RemoteCommandExecute func. Since this is a test function and we have some control of the environment I didn't think I needed to be any fancier than that.

Reply:
Alright. I wasn't originally aware of the fact that we get each command's stdout in a single string in the output array. This is ok.

Reply:
good to not use pipes if we can avoid it!